Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -178,52 +181,86 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
         return PlatformVersion(versionString)
     }
 
-    fileprivate func invoke(compiledExec: AbsolutePath, toolsVersion: ToolsVersion, writableDirectories: [AbsolutePath], input: Data) throws -> (outputJSON: Data, stdoutText: Data) {
-        // Construct the command line.
-        // FIXME: It would be more robust to pass it as `stdin` data, but we need TSC support for that.  When this is
-        // changed, PackagePlugin will need to change as well (but no plugins need to change).
-        var command = [compiledExec.pathString]
+    fileprivate func invoke(compiledExec: AbsolutePath, pluginArguments: [String], toolsVersion: ToolsVersion, writableDirectories: [AbsolutePath], input: Data) throws -> (outputJSON: Data, stdoutText: Data) {
+        // Construct the command line.  We just pass along any arguments intended for the plugin.
+        var command = [compiledExec.pathString] + pluginArguments
         command += [String(decoding: input, as: UTF8.self)]
 
         // If enabled, run command in a sandbox.
         // This provides some safety against arbitrary code execution when invoking the plugin.
         // We only allow the permissions which are absolutely necessary.
         if self.enableSandbox {
-            command = Sandbox.apply(command: command, writableDirectories: writableDirectories)
+            command = Sandbox.apply(command: command, writableDirectories: writableDirectories + [self.cacheDir])
         }
 
-        // Invoke the plugin script as a subprocess.
-        let result: ProcessResult
+        // Create and configure a Process.
+        let process = Process()
+        process.launchPath = command.first!
+        process.arguments = Array(command.dropFirst())
+        process.environment = ProcessInfo.processInfo.environment
+        process.currentDirectoryURL = self.cacheDir.asURL
+            
+        // Create a dispatch group for waiting until the process has terminated and the data has been read.
+        let waiters = DispatchGroup()
+        // Set up for capturing stdout data.
+        let stdoutPipe = Pipe()
+        var stdoutData = Data()
+        waiters.enter()
+        stdoutPipe.fileHandleForReading.readabilityHandler = { (fileHandle: FileHandle) -> Void in
+            let newData = fileHandle.availableData
+            if newData.isEmpty {
+                fileHandle.readabilityHandler = nil
+                waiters.leave()
+            }
+            else {
+                stdoutData.append(contentsOf: newData)
+            }
+        }
+        process.standardOutput = stdoutPipe
+        
+        // Set up for capturing stderr data.
+        waiters.enter()
+        let stderrPipe = Pipe()
+        var stderrData = Data()
+        stderrPipe.fileHandleForReading.readabilityHandler = { (fileHandle: FileHandle) -> Void in
+            let newData = fileHandle.availableData
```

Comment:
This is very similar to the one for `stdout` right now, but this will start to diverge as we handle JSON messages on `stdout` and freeform text on `stderr`.

In the future we could switch to gRPC or to using distributed actors, but a simple JSON protocol is sufficient during bringup and avoids pulling in other dependencies while until this stabilizes.